libkbfs/md_ops.go

```diff
@@ -421,13 +456,26 @@ func (md *MDOpsStandard) checkRevisionCameBeforeMerkle(
 	}
 
 	// Otherwise it's valid, as long as there's a valid chain of MD
-	// revisions between the two, and the global root info checks out
-	// as well.  By using `ctx`, we'll avoid infinite loops in the
+	// revisions between the two.  First, see if what chain we've
+	// already validated, and if this revision falls in that chain,
+	// we're done.  Otherwise, just fetch the part of the chain we
+	// haven't validated yet.
+	newChainEnd := md.startOfValidatedChainForLeaf(irmd.TlfID(), leaf.Revision)
+	if newChainEnd == kbfsmd.RevisionUninitialized {
+		newChainEnd = leaf.Revision
```

Comment:
WDYT just have `startOfValidatedChainForLeaf` return this, so we don't have to rely on the caller to remember to check if the returned revision is `kbfsmd.RevisionUninitialized`?

Reply:
Done.